examples/rl/rllib/rllib.py

```diff
@@ -126,69 +196,95 @@ def main(
         # Specifies additional mutations after hyperparam_mutations is applied
         custom_explore_fn=explore,
     )
+    # analysis = tune.run(
+    #     "PG",
+    #     name=experiment_name,
+    #     stop={"time_total_s": time_total_s},
+    #     checkpoint_freq=checkpoint_freq,
+    #     checkpoint_at_end=True,
+    #     local_dir=str(result_dir),
+    #     resume=resume_training,
+    #     restore=checkpoint,
+    #     max_failures=3,
+    #     num_samples=num_samples,
+    #     export_formats=["model", "checkpoint"],
+    #     config=algo_config,
+    #     scheduler=pbt,
+    # )
 
-    # XXX: There is a bug in Ray where we can only export a trained model if
-    #      the policy it's attached to is named 'default_policy'.
-    #      See: https://github.com/ray-project/ray/issues/5339
-    rllib_policies = {
-        "default_policy": (
-            None,
-            rllib_agent["observation_space"],
-            rllib_agent["action_space"],
-            {"model": {"custom_model": TrainingModel.NAME}},
-        )
-    }
+    # print(analysis.dataframe().head())
 
-    smarts.core.seed(seed)
-    tune_config = {
-        "env": RLlibHiWayEnv,
-        "log_level": "WARN",
-        "num_workers": num_workers,
-        "env_config": {
-            "seed": tune.sample_from(lambda spec: random.randint(0, 300)),
-            "scenarios": [str(Path(scenario).expanduser().resolve().absolute())],
-            "headless": not envision,
-            "agent_specs": {
-                f"AGENT-{i}": rllib_agent["agent_spec"] for i in range(num_agents)
-            },
-        },
-        "multiagent": {"policies": rllib_policies},
-        "callbacks": Callbacks,
-    }
+    # best_logdir = Path(analysis.get_best_logdir("episode_reward_max", mode="max"))
+    # model_path = best_logdir / "model"
 
-    experiment_name = "rllib_example_multi"
-    result_dir = Path(result_dir).expanduser().resolve().absolute()
-    if checkpoint_num:
-        checkpoint = str(
-            result_dir / f"checkpoint_{checkpoint_num}" / f"checkpoint-{checkpoint_num}"
-        )
-    else:
-        checkpoint = None
+    # copy_tree(str(model_path), save_model_path, overwrite=True)
+    # print(f"Wrote model to: {save_model_path}")
 
-    print(f"Checkpointing at {str(result_dir)}")
-    analysis = tune.run(
-        "PG",
-        name=experiment_name,
-        stop={"time_total_s": time_total_s},
-        checkpoint_freq=1,
-        checkpoint_at_end=True,
-        local_dir=str(result_dir),
-        resume=resume_training,
-        restore=checkpoint,
-        max_failures=3,
-        num_samples=num_samples,
-        export_formats=["model", "checkpoint"],
-        config=tune_config,
-        scheduler=pbt,
-    )
+    ## Approach 2
```

Comment:
This approach could be useful because it makes very evident what is happening with the training since very little is automated in terms of schenduling because it does not use `Tune`. The downside is that it does not get the trialing and hyperparameter manipulation of schedulers like the `PopulationBasedTraining` scheduler that was originally used.

I am not sure if we should completely abandon this approach since it is very clear how it works. I am considering splitting this approach out.